test/test_cpp_extensions_open_device_registration.py

```diff
@@ -259,7 +259,7 @@ def test_open_device_storage_pin_memory():
 
         def test_open_device_serialization():
             storage = torch.UntypedStorage(4, device=torch.device('foo'))
-            self.assertEqual(torch.serialization.location_tag(storage), 'foo')
+            self.assertEqual(torch.serialization.location_tag(storage), 'foo:0')
```

Comment:
The device index is not required. so this shouldn't be needed.

Reply:
@albanD 
In actual case, the device index is not necessary. In `location_tag` method, device index of storage  is checked.
```
if obj.device.index is None:
    return backend_name
else:
    return backend_name + ':' + str(obj.device.index)
```

This test case creates `c10::Device(c10::DeviceType::PrivateUse1, 0)` as a fixed return in open_registration_extension.cpp . Therefore, the `location_tag` result contains device index.

Reply:
Could we keep both then?

Reply:
Fix the bug in [#100632](https://github.com/pytorch/pytorch/pull/100632), please review.
I will resolve conflicts in this PR.